packages/core/src/utils/parseInvitation.ts

```diff
@@ -1,10 +1,122 @@
+  // eslint-disable-next-line no-restricted-globals
+  const abortController = new AbortController()
+  const id = setTimeout(() => abortController.abort(), 15000)
+  let response
+  try {
+    response = await dependencies.fetch(invitationUrl, {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    })
+  } catch (error) {
+    throw new AriesFrameworkError('Get request failed on provided Url')
+  }
+  clearTimeout(id)
+  return response
+}
+
+//This currently does not follow the RFC because of issues with fetch, currently uses a janky work around
+export const fromShortUrl = async (response: Response): Promise<OutOfBandInvitation> => {
+  if (response) {
+    if (response.headers.get('Content-Type') === 'application/json' && response.ok) {
+      const invitationJson = await response.json()
+      if (invitationJson['@type'].includes('out-of-band')) {
+        const invitation = JsonTransformer.fromJSON(invitationJson, OutOfBandInvitation)
+
+        await MessageValidator.validateSync(invitation)
+
+        return invitation
+      } else {
+        const invitation = JsonTransformer.fromJSON(invitationJson, ConnectionInvitationMessage)
+
+        await MessageValidator.validateSync(invitation)
+
+        return convertToNewInvitation(invitation)
+      }
+    } else if (response['url']) {
+      // The following if else is for here for trinsic shorten urls
+      // Because the redirect targets a deep link the automatic redirect does not occur
+      let parsedUrl
+      const location = response.headers.get('Location')
+      if ((response.status === 302 || response.status === 301) && location) parsedUrl = parseUrl(location).query
+      else parsedUrl = parseUrl(response['url']).query
+
+      if (parsedUrl['oob']) {
+        const encodedInvitation = parsedUrl['oob']
+        let invitationJson = null
+        if (typeof encodedInvitation === 'string') invitationJson = JsonEncoder.fromBase64(encodedInvitation)
+        const invitation = JsonTransformer.fromJSON(invitationJson, OutOfBandInvitation)
+
+        await MessageValidator.validateSync(invitation)
+
+        return invitation
+      } else {
+        if (parsedUrl['c_i'] || parsedUrl['d_m']) {
+          const encodedInvitation = parsedUrl['c_i'] ?? parsedUrl['d_m']
+          let invitationJson = null
+          if (typeof encodedInvitation === 'string') invitationJson = JsonEncoder.fromBase64(encodedInvitation)
+          const invitation = JsonTransformer.fromJSON(invitationJson, ConnectionInvitationMessage)
+
+          await MessageValidator.validateSync(invitation)
+
+          return convertToNewInvitation(invitation)
+        }
+        throw new AriesFrameworkError(
+          'InvitationUrl is invalid. Needs to be encrypted with either c_i, d_m, or oob or must be valid shortened URL'
+        )
+      }
```

Comment:
Can't we use the already available `parseInvitationUrl` for this?

Reply:
Yes, that is a good point